language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/diagnostic/DiagnosticsHelper.java

```diff
@@ -77,25 +77,27 @@ public synchronized void compileAndSendDiagnostics(ExtendedLanguageClient client
         if (project.isEmpty()) {
             return;
         }
-        Map<String, List<Diagnostic>> diagnosticMap = getLatestDiagnostics(context);
+        Map<String, List<Diagnostic>> latestDiagnostics = getLatestDiagnostics(context);
 
         // If the client is null, returns
         if (client == null) {
             return;
         }
+        Map<String, List<Diagnostic>> lastProjectDiagnostics =
+                lastDiagnosticMap.getOrDefault(project.get().sourceRoot(), new HashMap<>());
 
-        // Clear old entries with an empty list
-        lastDiagnosticMap.forEach((key, value) -> {
-            if (!diagnosticMap.containsKey(key)) {
+        // Clear old diagnostic entries of the project with an empty list
+        lastProjectDiagnostics.forEach((key, value) -> {
+            if (!latestDiagnostics.containsKey(key)) {
                 client.publishDiagnostics(new PublishDiagnosticsParams(key, emptyDiagnosticList));
             }
         });
 
-        // Publish diagnostics
-        diagnosticMap.forEach((key, value) -> client.publishDiagnostics(new PublishDiagnosticsParams(key, value)));
+        // Publish diagnostics for the project
+        latestDiagnostics.forEach((key, value) -> client.publishDiagnostics(new PublishDiagnosticsParams(key, value)));
 
-        // Replace old map
-        lastDiagnosticMap = diagnosticMap;
+        // Replace old diagnostic map associated with the project
+        lastDiagnosticMap.put(project.get().sourceRoot(), latestDiagnostics);
```

Comment:
Is this a mistake?

Reply:
I believe this was added because I fetched from main branch to the PR branch and this must be a diff between main and jbal-semtype

Reply:
Shall we take the updates via separate PRs only doing syncs?

Reply:
reverted merge